src/index.ts

```diff
@@ -89,14 +90,13 @@ const remarkEmbedder: Plugin<[RemarkEmbedderOptions]> = ({
         node.type === 'link' &&
         !node.title &&
         node.children.length === 1 &&
-        node.children[0].value === node.url
-      if (!isText && !isValidLink) {
+        (node.children[0] as Text).value === node.url
```

Comment:
@wooorm
`Link`'s `children` are typed as `StaticPhrasingContent[]`, but I think they're actually `[Text]`, no?

Reply:
No, emphasis, images, inline code and such, all also can exist in links. Just not other links

Reply:
@wooorm how would they look like then when putting them in Markdown?

Reply:
```
[*emphasis*](https://example.com)
[**strong**](https://example.com)
[![image]()](https://example.com)
...
```

Reply:
As there's always only 1 child, the type should be `[StaticPhrasingContent]` instead of `StaticPhrasingContent[]`, no?

Reply:
There is not always one type. Things can exist together in links: `[*emphasis* text **strong** text ![image](#) ~delete~](https://example.com)`

Reply:
> As there's always only 1 child, the type should be [StaticPhrasingContent] instead of StaticPhrasingContent[], no?

`[**strong** text](https://example.com)` is a use-case of having multiple children, my bad 😅